gradio/pipelines.py

```diff
@@ -3,268 +3,77 @@
+        pipeline_info = _handle_transformers_pipeline(pipeline)
+
+    elif str(type(pipeline)).startswith("<class 'diffusers.pipelines"):
+        pipeline_info = _handle_diffusers_pipeline(pipeline)
+
     else:
-        raise ValueError(f"Unsupported pipeline type: {type(pipeline)}")
+        raise ValueError(
+            "pipeline must be a transformers.pipeline or diffusers.pipeline"
+        )
 
     # define the function that will be called by the Interface
     def fn(*params):
         data = pipeline_info["preprocess"](*params)
-        # special cases that needs to be handled differently
-        if isinstance(
-            pipeline,
-            (
-                pipelines.text_classification.TextClassificationPipeline,
-                pipelines.text2text_generation.Text2TextGenerationPipeline,
-                pipelines.text2text_generation.TranslationPipeline,
-            ),
-        ):
-            data = pipeline(*data)
-        else:
+        if "tranformers.pipelines" in str(type(pipeline)):
+            from transformers import pipelines
+
+            # special cases that needs to be handled differently
+            if isinstance(
+                pipeline,
+                (
+                    pipelines.text_classification.TextClassificationPipeline,
+                    pipelines.text2text_generation.Text2TextGenerationPipeline,
+                    pipelines.text2text_generation.TranslationPipeline,
+                ),
+            ):
+                data = pipeline(*data)
+            else:
+                data = pipeline(**data)
+            # special case for object-detection
+            # original input image sent to postprocess function
+            if isinstance(
+                pipeline,
+                pipelines.object_detection.ObjectDetectionPipeline,
+            ):
+                output = pipeline_info["postprocess"](data, params[0])
+            else:
+                output = pipeline_info["postprocess"](data)
+            return output
+
+        elif str(type(pipeline)).startswith("<class 'diffusers.pipelines"):
             data = pipeline(**data)
-        # special case for object-detection
-        # original input image sent to postprocess function
-        if isinstance(
-            pipeline,
-            pipelines.object_detection.ObjectDetectionPipeline,
-        ):
-            output = pipeline_info["postprocess"](data, params[0])
-        else:
             output = pipeline_info["postprocess"](data)
-        return output
+            return output
 
     interface_info = pipeline_info.copy()
     interface_info["fn"] = fn
     del interface_info["preprocess"]
     del interface_info["postprocess"]
 
     # define the title/description of the Interface
-    interface_info["title"] = pipeline.model.__class__.__name__
+    interface_info["title"] = (
+        pipeline.model.__class__.__name__
+        if str(type(pipeline)).startswith("<class 'transfomers.pipelines")
```

Comment:
Typo here too.

Reply:
Fixed it for now, but i will add tests for this to make sure this does not happen again.